From 67b0d49ccd0fd7bb3a14fe8929d813429ddb7660 Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Sat, 3 Aug 2024 12:09:28 +0530
Subject: [PATCH] Updated media c2 hal from hidl to aidl.

Updated android.hardware.media.c2 hal from hidl V1.0 to aidl V1.
Adding media c2 software interface in device compatibility matrix.
Adding venodr prop for media c2 to enable aidl hal.

Tests: Prepared EB and its booted successfully.

Tracked-On: OAM-122196
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 groups/device-specific/caas/compatibility_matrix.xml | 4 ++++
 groups/device-specific/caas/product.mk               | 4 +++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/groups/device-specific/caas/compatibility_matrix.xml b/groups/device-specific/caas/compatibility_matrix.xml
index 6b86d07..8f935e8 100644
--- a/groups/device-specific/caas/compatibility_matrix.xml
+++ b/groups/device-specific/caas/compatibility_matrix.xml
@@ -3,6 +3,10 @@
         <name>android.frameworks.sensorservice</name>
         <fqname>ISensorManager/default</fqname>
     </hal>
+    <hal format="aidl">
+        <name>android.hardware.media.c2</name>
+        <fqname>IComponentStore/software</fqname>
+    </hal>
     <!--hal format="hidl" optional="false">
         <name>android.frameworks.sensorservice</name>
         <version>1.0</version>
diff --git a/groups/device-specific/caas/product.mk b/groups/device-specific/caas/product.mk
index 6dc8986..c498905 100755
--- a/groups/device-specific/caas/product.mk
+++ b/groups/device-specific/caas/product.mk
@@ -29,7 +29,7 @@ PRODUCT_PACKAGES +=  \
                     android.hardware.graphics.allocator-service.minigbm \
                     android.hardware.identity \
                     android.hardware.identity-service.example \
-		    android.hardware.graphics.composer3-service.intel \
+                    android.hardware.graphics.composer3-service.intel \
                     android.hardware.media.omx@1.0-service
 
 
@@ -69,3 +69,5 @@ PRODUCT_COPY_FILES += device/intel/common/host_scripts/start_flash_usb.sh:$(PROD
 PRODUCT_COPY_FILES += vendor/intel/fw/trusty-release-binaries/rpmb_dev:$(PRODUCT_OUT)/scripts/rpmb_dev
 PRODUCT_COPY_FILES += $(LOCAL_PATH)/wakeup.py:$(PRODUCT_OUT)/scripts/wakeup.py
 PRODUCT_COPY_FILES += device/intel/civ/host/vm-manager-binaries/vm-manager.deb:$(PRODUCT_OUT)/scripts/vm-manager.deb
+
+PRODUCT_VENDOR_PROPERTIES += media.c2.hal.selection=aidl
-- 
2.45.2

